assignment4.Rmd

```diff
@@ -114,3 +117,34 @@ biplot(pca)
 ```
```

Comment:
You can throw the loadings into a DF to show the proportionality of the variables for each component by running the below:
D5 <- mutate_all(as.data.frame(loadings), funs((.)/sum(.)))

This can help you to create the composite variables that allow you to weigh each variable from the original data frame.

D2 <- as.data.frame(D2)
D2$aggressiveness <- D2$prior_prob_count*0.12423113 + D2$prior_percent_correct*0.08035956 + D2$problems_attempted*0.21744737 + D2$mean_hint*0.30223780 + D2$mean_attempt*0.25863458 + D2$mean_confidence*0.01708956